### PR TITLE
Clean old data to be GDPR compliant

### PR DESCRIPTION
**Pull request by Dcoutelle**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N 
| New feature? | Y
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/287
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5878
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Clean data in the database according to European GDPR regulation.

#### Steps to test this PR:
1. Take 4 leads, with pages_hit history
2. In database, for lead 1, modify `last_active` to 2014 and set `date_identified` to 2014 _(identifed lead)_
3. In database, for lead 2, modify `last_active` to 2014 and set `date_identified` to `null` _(anonymous lead)_
4. In database, for lead 3, modify `last_active` to 2018 and set `date_identified` to `null` _(anonymous lead)_
5. In database, for lead 5, modify `last_active` to `null`, set `date_identified` to `null` and `date_added` to 2014 (_simulate imported lead_)
5. In database, for lead 5, modify `last_active` to `null`, set `date_identified` to `null` and `date_added` to 2018 (_simulate imported lead_)

6. run `mautic:maintenance:cleanup --gdpr`
7. lead 1, 2, 5 must be deleted for database with associed pageHits

### CODE REVIEW

**Comment by heathdutton** [comment]
Looks good to me. 👍

**Review by npracht — APPROVED** [review]
👍

**Comment by npracht** [comment]
@heathdutton can you revalidate with the review system and tag ready to commit if still ok ?

**Comment by Woeler** [comment]
Can anyone inform on this GDPR requirement? Is it about deleting old data? Is there a limit?

File: app/bundles/CoreBundle/Command/CleanupMaintenanceCommand.php [hunk 1]
```diff
@@ -42,6 +42,7 @@ protected function configure()
                         365
                     ),
                     new InputOption('dry-run', 'r', InputOption::VALUE_NONE, 'Do a dry run without actually deleting anything.'),
+                    new InputOption('gdpr', 'g', InputOption::VALUE_NONE, 'Delete data to fullfil GDPR requiement.'),
```
Comment (Woeler) [inline app/bundles/CoreBundle/Command/CleanupMaintenanceCommand.php]: typo *requirement

Comment (npracht) [reply app/bundles/CoreBundle/Command/CleanupMaintenanceCommand.php]: done

**Review by Woeler — APPROVED** [review]
This works great, tested it locally.

**Pull request merged**